File: Vehicle_ClaimApp/FinalSubmitClaimApp/app/src/main/java/com/example/vehicleinsuranceclaimapp/fragment/RegisterFragment.java
package com.example.vehicleinsuranceclaimapp.fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.vehicleinsuranceclaimapp.R;
import com.example.vehicleinsuranceclaimapp.dao.UserDao;
import com.example.vehicleinsuranceclaimapp.database.AppDatabase;
import com.example.vehicleinsuranceclaimapp.model.User;

import java.util.concurrent.Executors;

public class RegisterFragment extends Fragment {
    private EditText usernameEditText, emailEditText,  passwordEditText;
    private Button registerButton;
    private TextView loginBtn;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_register, container, false);

        usernameEditText = view.findViewById(R.id.usernameEditText);
        passwordEditText = view.findViewById(R.id.passwordEditText);
        emailEditText = view.findViewById(R.id.emailEditText);
        registerButton = view.findViewById(R.id.registerButton);
        loginBtn = view.findViewById(R.id.loginBtn);

        registerButton.setOnClickListener(v ->{
            String username = usernameEditText.getText().toString().trim();
            String email = emailEditText.getText().toString().trim();
            String password = passwordEditText.getText().toString().trim();

            if(TextUtils.isEmpty(username) || TextUtils.isEmpty(email) || TextUtils.isEmpty(password)){
                Toast.makeText(getContext(), "Please fill all details", Toast.LENGTH_SHORT).show();

                return;
            }

            Executors.newSingleThreadExecutor().execute(() -> {
                AppDatabase db = AppDatabase.getInstance(getContext());
                UserDao userDao = db.userDao();

                if(userDao.checkIfUserExists(username,email) != null){
                    getActivity().runOnUiThread(() -> {
                        Toast.makeText(getContext(), "User already exists", Toast.LENGTH_SHORT).show();
                    });
                }else{
                    userDao.insertUser(new User(username, email, password));
                    getActivity().runOnUiThread(() -> {
                        Toast.makeText(getContext(), "User Registered successfully", Toast.LENGTH_SHORT).show();
                        requireActivity().getSupportFragmentManager().popBackStack();
                    });
                }
            });
        });

        loginBtn.setOnClickListener(v -> navigateToLoginFragment());


        return view;
    }


    private void navigateToLoginFragment() {

        LoginFragment loginFragment = new LoginFragment();
        requireActivity().getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragmentContainer, loginFragment)
                .addToBackStack(null)
                .commit();
    }
}

File: BookMySlot/app/src/main/java/com/example/bookmyslot/ui/RegisterFragment.kt
package com.example.bookmyslot.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import com.example.bookmyslot.R
import com.example.bookmyslot.database.AppDatabase
import com.example.bookmyslot.databinding.FragmentRegisterBinding
import com.example.bookmyslot.model.User
import kotlinx.coroutines.launch
import java.util.concurrent.Executors

class RegisterFragment : Fragment() {

    private lateinit var binding: FragmentRegisterBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentRegisterBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.registerButton.setOnClickListener {
            registerUser()
        }

        binding.loginBtn.setOnClickListener {
            findNavController().navigate(R.id.action_registerFragment_to_loginFragment)
        }
    }

    //function to register the new user and check if the user is already registered or not
    private fun registerUser() {
        val username = binding.usernameEditText.text.toString().trim()
        val email = binding.emailEditText.text.toString().trim()
        val password = binding.passwordEditText.text.toString().trim()

        if (username.isEmpty() || email.isEmpty() || password.isEmpty()) {
            Toast.makeText(requireContext(), "Please fill all details", Toast.LENGTH_SHORT).show()
            return
        }

        lifecycleScope.launch {
            val db = AppDatabase.getInstance(requireContext())
            val userDao = db.userDao()
            val user = User(username = username, email = email, password = password)

            if (userDao.checkIfUserExists(username, email) != null) {
                requireActivity().runOnUiThread {
                    Toast.makeText(requireContext(), "User already exists", Toast.LENGTH_SHORT).show()
                }
            } else {
                userDao.insertUser(user)
                requireActivity().runOnUiThread {
                    Toast.makeText(requireContext(), "User Registered successfully", Toast.LENGTH_SHORT).show()
                    findNavController().navigateUp()
                }
            }
        }
    }
}